crates/shared/src/recent_block_cache.rs

```diff
@@ -304,7 +309,7 @@ where
             match self.cached_most_recently_at_block.entry(key.clone()) {
                 Entry::Occupied(mut entry) => {
                     let value = entry.get_mut();
-                    *value = (*value).max(block);
+                    *value = cmp::max(*value, block);
```

Comment:
This was weird... for some reason `(*value)` was being inferred as an `Interator` and then it was complaining that `.max()` expects 0 arguments (as it was the `Iterator::max`).